GitVersionTask fails when building from a tag

**Describe the bug**
I'm using GitHub actions to create a new build of a project that is using GitVersionTask.

When the build runs on a branch, everything works fine. When it runs for a tag, the following error is outputted.

```
/root/.nuget/packages/gitversiontask/5.3.5/build/GitVersionTask.targets(46,9): warning : 2020-05-31 05:47:23                WARN [05/31/20 5:47:23:62] Failed to inherit 
Increment branch configuration, no branches found. [/src/Core/Core.csproj]
/root/.nuget/packages/gitversiontask/5.3.5/build/GitVersionTask.targets(46,9): warning : Falling back to master branch config [/src/Core/Core.csproj]
/root/.nuget/packages/gitversiontask/5.3.5/build/GitVersionTask.targets(46,9): warning :  [/src/Core/Core.csproj]
/root/.nuget/packages/gitversiontask/5.3.5/build/GitVersionTask.targets(46,9): warning : It looks like the branch being examined is a detached Head pointing to commit '87436ac'. Without a proper branch name GitVersion cannot determine the build version. [/src/Core/Core.csproj]
/root/.nuget/packages/gitversiontask/5.3.5/build/GitVersionTask.targets(10,9): warning : 2020-05-31 05:47:24                WARN [05/31/20 5:47:24:60] Failed to inherit 
Increment branch configuration, no branches found. [/src/Core/Core.csproj]
/root/.nuget/packages/gitversiontask/5.3.5/build/GitVersionTask.targets(10,9): warning : Falling back to master branch config [/src/Core/Core.csproj]
/root/.nuget/packages/gitversiontask/5.3.5/build/GitVersionTask.targets(10,9): warning :  [/src/Core/Core.csproj]
/root/.nuget/packages/gitversiontask/5.3.5/build/GitVersionTask.targets(10,9): warning : It looks like the branch being examined is a detached Head pointing to commit '87436ac'. Without a proper branch name GitVersion cannot determine the build version. [/src/Core/Core.csproj]
/root/.nuget/packages/gitversiontask/5.3.5/build/GitVersionTask.targets(71,9): warning : 2020-05-31 05:47:24                WARN [05/31/20 5:47:24:68] Failed to inherit 
Increment branch configuration, no branches found. [/src/Core/Core.csproj]
/root/.nuget/packages/gitversiontask/5.3.5/build/GitVersionTask.targets(71,9): warning : Falling back to master branch config [/src/Core/Core.csproj]
/root/.nuget/packages/gitversiontask/5.3.5/build/GitVersionTask.targets(71,9): warning :  [/src/Core/Core.csproj]
/root/.nuget/packages/gitversiontask/5.3.5/build/GitVersionTask.targets(71,9): warning : It looks like the branch being examined is a detached Head pointing to commit '87436ac'. Without a proper branch name GitVersion cannot determine the build version. [/src/Core/Core.csproj]
/usr/share/dotnet/sdk/2.2.207/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.GenerateAssemblyInfo.targets(161,5): error MSB4044: The "GetAssemblyVersion" task was not given a value for the required parameter "NuGetVersion". [/src/Core/Core.csproj]
/usr/share/dotnet/sdk/2.2.207/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.GenerateAssem
```

GitVersion standalone works fine when checking out the tag.

## Expected Behavior
The build completes with the correct version as per the tag

## Actual Behavior
GitVersionTask fails with the above error

## Steps to Reproduce
Run a build of a .NET Core 2.2 project in Docker using GitVersionTask 3.3.5

## Context
Unable to complete build

## Your Environment
GitVersionTask 5.3.5
Build running in a Docker container using mcr.microsoft.com/dotnet/core/sdk:2.2

GitVersion.yml

```
assembly-versioning-scheme: MajorMinorPatchTag
assembly-file-versioning-scheme: MajorMinorPatchTag
mode: ContinuousDeployment
continuous-delivery-fallback-tag: ""
branches:
  master:
    mode: ContinuousDeployment
    tag: "ci"
    increment: Minor
    track-merge-target: true
    tracks-release-branches: true
    is-release-branch: false
    prevent-increment-of-merged-branch-version: false
  release:
    regex: release?[/]
    mode: ContinuousDeployment
    increment: Patch
    is-release-branch: true
    prevent-increment-of-merged-branch-version: true
ignore:
  sha: []
```

--- a/src/GitVersionCore.Tests/IntegrationTests/TagCheckoutScenarios.cs
+++ b/src/GitVersionCore.Tests/IntegrationTests/TagCheckoutScenarios.cs
@@ -1,0 +1,32 @@
+﻿using GitTools.Testing;
+using NUnit.Framework;
+
+namespace GitVersionCore.Tests.IntegrationTests
+{
+    [TestFixture]
+    public class TagCheckoutScenarios
+    {
+        [Test]
+        public void GivenARepositoryWithSingleCommit()
+        {
+            using var fixture = new EmptyRepositoryFixture();
+            const string taggedVersion = "1.0.3";
+            fixture.Repository.MakeATaggedCommit(taggedVersion);
+            fixture.Checkout(taggedVersion);
+
+            fixture.AssertFullSemver(taggedVersion);
+        }
+
+        [Test]
+        public void GivenARepositoryWithSingleCommitAndSingleBranch()
+        {
+            using var fixture = new EmptyRepositoryFixture();
+            const string taggedVersion = "1.0.3";
+            fixture.Repository.MakeATaggedCommit(taggedVersion);
+            fixture.BranchTo("task1");
+            fixture.Checkout(taggedVersion);
+
+            fixture.AssertFullSemver(taggedVersion);
+        }
+    }
+}

--- a/src/GitVersionCore.Tests/IntegrationTests/TagCheckoutScenarios.cs
+++ b/src/GitVersionCore.Tests/IntegrationTests/TagCheckoutScenarios.cs
@@ -1,4 +1,4 @@
-﻿using GitTools.Testing;
+using GitTools.Testing;
 using NUnit.Framework;
 
 namespace GitVersionCore.Tests.IntegrationTests

--- a/src/GitVersionCore/VersionCalculation/NextVersionCalculator.cs
+++ b/src/GitVersionCore/VersionCalculation/NextVersionCalculator.cs
@@ -35,7 +35,10 @@ namespace GitVersion.VersionCalculation
                 log.Info($"Current commit is tagged with version {context.CurrentCommitTaggedVersion}, " +
                          "version calculation is for metadata only.");
             }
-            EnsureHeadIsNotDetached(context);
+            else
+            {
+                EnsureHeadIsNotDetached(context);
+            }
 
             SemanticVersion taggedSemanticVersion = null;
 

